src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -352,31 +352,19 @@ CHIP_ERROR UDPEndPointImplLwIP::GetPCB(IPAddressType addrType)
 void UDPEndPointImplLwIP::LwIPReceiveUDPMessage(void * arg, struct udp_pcb * pcb, struct pbuf * p, const ip_addr_t * addr,
                                                 u16_t port)
 {
-    UDPEndPointImplLwIP * ep       = static_cast<UDPEndPointImplLwIP *>(arg);
-    IPPacketInfo * pktInfo         = nullptr;
-    System::PacketBufferHandle buf = System::PacketBufferHandle::Adopt(p);
-
+    UDPEndPointImplLwIP * ep = static_cast<UDPEndPointImplLwIP *>(arg);
+    IPPacketInfo * pktInfo   = nullptr;
     if (ep->mState == State::kClosed)
-        return;
-
-    if (buf->HasChainedBuffer())
     {
-        // Try the simple expedient of flattening in-place.
-        buf->CompactHead();
+        pbuf_free(p);
```

Comment:
could you have some form of RAII? adding pbuf_free at various places is not easy too maintain/follow.